packages/@ngtools/benchmark/src/utils.ts

```diff
@@ -0,0 +1,177 @@
+    }
+    return;
+  };
+  let all: any[] = [];
+  for (let i = 0; i < array.length; i++) {
+    fn(i, array, [], all);
+  }
+  if (array.length > 0) { all.push(array); }
+  return all;
+}
+
+// Returns a promise with the result of calling `times` times the `fn` function with `args`
+// `fn` will be called in parallel, and is expected to return a promise
+// Not used currently
+export function parallelMultiPromise(times: number, fn: any, ...args: string[]) {
+  return Promise.all(makeArray(times).map(() => fn.apply(null, args)));
+}
+
+// Returns a promise with the result of calling `times` times the `fn` function with `args`
+// `fn` will be called in serial, and is expected to return a promise
+export function serialMultiPromise(times: number, fn: any, ...args: any[]) {
+  let results: Promise<any>[] = [];
+  let promise = Promise.resolve();
+  makeArray(times).forEach(() => promise = promise.then(() =>
+    fn.apply(null, args).then((result: Promise<any>) => results.push(result))
+  ));
+  return promise.then(() => results);
+}
+
+// Spawns `cmd` with `args`
+// Calls `matchFn` with the `stdout` output, a `result` var and the process
+// `dataFn` is expected to modify `result`
+export function matchSpawn(debug: boolean, cwd: string, matchFn: any,
+  cmd: string, args: string[] = []) {
+  // dataFn will have access to result and use it to store results
+  let result = {
+    // overrideErr will signal that an error code on the exit event should be ignored
+    // This is useful on windows where killing a tree of processes always makes them
+    // exit with an error code
+    overrideErr: false,
+    time: 0
+  };
+  let stdout = '';
+  let stderr = '';
+  let spawnOptions: any = {
+    cwd: cwd,
+    env: process.env
+  };
+
+  // Split given `cmd` and pass on any args to `args`
+  const splitCmd = cmd.split(' ');
+  cmd = splitCmd.shift();
+  args = splitCmd.concat(args);
+
+  if (process.platform.startsWith('win')) {
+    args = ['/c', cmd].concat(args);
+    cmd = 'cmd.exe';
+    spawnOptions['stdio'] = 'pipe';
+  }
+
+  if (debug) {
+    console.log(`spawning cmd: ${cmd}, args: ${args}, cwd: ${cwd}`);
+  }
+
+  const startTime = Date.now();
+  const childProcess = child_process.spawn(cmd, args, spawnOptions);
+
+  childProcess.stdout.on('data', (data: Buffer) => {
+    let strippedData = stripAnsi(data.toString('utf-8'));
+    if (debug) {
+      console.log(strippedData);
+    }
+    stdout += strippedData;
+    if (matchFn) {
+      matchFn(strippedData, result, childProcess);
+    }
+  });
+
+  childProcess.stderr.on('data', (data: Buffer) => {
+    let strippedData = stripAnsi(data.toString('utf-8'));
+    if (debug) {
+      console.log(strippedData);
+    }
+    stderr += strippedData;
+  });
+
+  return new Promise((resolve, reject) =>
+    childProcess.on('exit', (err: number) => {
+      result.time = Date.now() - startTime;
+      return err && !result.overrideErr
+        ? resolve({ err, stdout, stderr })
```

Comment:
No, still I since want to gather the remaining results. If you're doing a lot of iterations you don't want to throw everything away just because one failed.